tests/apollo/test_skvbc_view_change.py

```diff
@@ -318,17 +318,16 @@ class SkvbcViewChangeTest(unittest.TestCase):
 
         bft_network.start_all_replicas()
 
-        initial_primary = 0
-        
-        replcas_to_stop = [ v for v in range(initial_primary,
-                                             initial_primary + num_consecutive_failing_primaries) ]
-        
-        expected_final_primary = initial_primary + num_consecutive_failing_primaries
+        initial_primary = await bft_network.get_current_primary()
+        replcas_to_stop = [ v for v in range((initial_primary % bft_network.config.n),
+                                             (initial_primary % bft_network.config.n) + num_consecutive_failing_primaries) ]
```

Comment:
The logic here is wrong on several levels. First of all, `initial_primary` is always in the interval `[0,n]`, as you have `initial_primary = await bft_network.get_current_primary()`.

Also, a range will not work for what you want to achieve here, because you want potentially to have a list of primaries, such as [5,0,1,2].

Please review this logic...

Reply:
Teo and I agreed that we need to have 2 methods, one to get the current primary and one to get the current view.